xbmc/dialogs/GUIDialogMediaFilter.h

```diff
@@ -44,9 +44,9 @@ class CGUIDialogMediaFilter : public CGUIDialogSettingsManualBase
     std::string controlType;
     std::string controlFormat;
     CDatabaseQueryRule::SEARCH_OPERATOR ruleOperator;
-    std::shared_ptr<CSetting> setting;
-    CSmartPlaylistRule *rule;
-    void *data;
+    std::shared_ptr<CSetting> setting = nullptr;
```

Comment:
This line was fine before, objects are default-constructed.

Reply:
I don't see what mean?

Reply:
This is the error
```
/home/lukas/Documents/git/xbmc/xbmc/dialogs/GUIDialogMediaFilter.cpp:132:1: warning: missing initializer for member ‘CGUIDialogMediaFilter::Filter::setting’ [-Wmissing-field-initializers]
  132 | };
      | ^
/home/lukas/Documents/git/xbmc/xbmc/dialogs/GUIDialogMediaFilter.cpp:132:1: warning: missing initializer for member ‘CGUIDialogMediaFilter::Filter::rule’ [-Wmissing-field-initializers]
/home/lukas/Documents/git/xbmc/xbmc/dialogs/GUIDialogMediaFilter.cpp:132:1: warning: missing initializer for member ‘CGUIDialogMediaFilter::Filter::data’ [-Wmissing-field-initializers]
<snip>
```
It's because `CGUIDialogMediaFilter::Filter` doesn't get default initialized, so we either have to set initial values or set them when all the other values are set.

Reply:
@garbear is this ok with you or do you have a better idea?